### ISSUE TITLE
Resnet baseline result problem ?

### ISSUE THREAD

**Issue by Xingxiangrui** [chenzhaomin123/ML_GCN#13]
In your paper, ResNet-101 result on COCO dataset is mAP 77.3  .  But we realize ResNet-101 on COCO dataset and achieves far better result which achieves mAP 80.557.  Is the result in your paper run by yourself or cited on the other paper?  If the result run by yourself ,  could you please release the ResNet-101 code that you use ?

**Comment by Xingxiangrui** [comment]
1. ResNet101 structure，and global max pooling( not global average pooling)
2. Loss function in this paper:  Multi-attribute Learning for Pedestrian Attribute Recognition in Surveillance Scenarios
In which :





------------------ 原始邮件 ------------------
发件人: "wp8619"<notifications@github.com>; 
发送时间: 2019年5月8日(星期三) 晚上8:03
收件人: "chenzhaomin123/ML_GCN"<ML_GCN@noreply.github.com>; 
抄送: "邢翔瑞"<1057945230@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [chenzhaomin123/ML_GCN] Resnet baseline result problem ? (#13)




Can I ask which paper reports vanilla rsnet101 can achieve 80.557? In the following paper, 75.2 was reported.
 
https://arxiv.org/pdf/1702.05891.pdf

**Comment by cgpeter96** [comment]
@Xingxiangrui   hello，我最近也用resnet101跑了一遍，发现mAP可以到82.288，代码如下。GCN的模块貌似没什么作用
```
class ResnetBaseFeature(nn.Module):
    def __init__(self, model):
        super().__init__()
        self.features = nn.Sequential(
            model.conv1,
            model.bn1,
            model.relu,
            model.maxpool,
            model.layer1,
            model.layer2,
            model.layer3,
            model.layer4,
        )
        self.pooling = nn.MaxPool2d(14, 14)

    def forward(self, x):
        feature = self.features(x)
        feature = self.pooling(feature)
        feature = feature.view(feature.size(0), -1)
        return feature

class ResNetM(nn.Module):
    def __init__(self,model,num_classes):
        super().__init__()
        self.features = ResnetBaseFeature(model)
        self.fc = nn.Linear(2048,num_classes)
        nn.init.kaiming_uniform_(self.fc.weight)

        self.image_normalization_mean = [0.485, 0.456, 0.406]
        self.image_normalization_std = [0.229, 0.224, 0.225]

    def forward(self,x,y):
        features =  self.features(x)
        return self.fc(features)
    def get_config_optim(self, lr, lrp):
        return [
            {'params': self.features.parameters(), 'lr': lr * lrp},
            {'params': self.fc.parameters(), 'lr': lr},
        ]

```

**Comment by chenzhaomin123** [comment]
@cgpeter96 你这baseline怎么刷的，我在跑实验的时候没有跑到你这个点，能具体分享一下怎么训练的吗？

你看我论文中描述：
Firstly, as the embedding-to-classiﬁer mapping parameters are shared across all classes, the learned classiﬁers can retain the weak semantic structures in the word embedding space, where semantic related concepts are close to each other. Meanwhile, the gradients of all classiﬁers can impact the classiﬁer generation function, which implicitly models the label dependencies.

还有后面的ablation study结果，在使用单位矩阵的时候我在coco上也能跑到81.7的结果。
即使不加边在传梯度的时候也会把隐含的关系建模进去，加边在我这边能有1个点左右的提升。

至于你随机产生邻接矩阵这个实验我确实没做过，能跑到comparable的结果可能由于你的矩阵很稀疏，几乎等于单位矩阵。
